images/virtualization-artifact/pkg/controller/dvcr-maintenance/condition/deployment.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2025 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package condition
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	dvcr_deployment_condition "github.com/deckhouse/virtualization/api/core/v1alpha2/dvcr-deployment-condition"
+)
+
+func NewMaintenanceCondition(reason dvcr_deployment_condition.MaintenanceReason, msgf string, args ...any) appsv1.DeploymentCondition {
+	status := "Unknown"
+	switch reason {
+	case dvcr_deployment_condition.LastResult:
+		status = "False"
+	case dvcr_deployment_condition.InProgress:
+		status = "True"
+	}
+
+	return appsv1.DeploymentCondition{
+		Type:           dvcr_deployment_condition.MaintenanceType,
+		Status:         corev1.ConditionStatus(status),
+		LastUpdateTime: metav1.Now(),
+		Reason:         string(reason),
+		Message:        fmt.Sprintf(msgf, args...),
+	}
+}
+
+// UpdateMaintenanceCondition replaces or removes Maintenance condition from deployment status.
+// Return true if status was changed.
+func UpdateMaintenanceCondition(deploy *appsv1.Deployment, reason dvcr_deployment_condition.MaintenanceReason, msgf string, args ...any) {
+	if deploy == nil {
+		return
+	}
+
+	condition := NewMaintenanceCondition(reason, msgf, args...)
+
+	// Condition is nil, so remove maintenance condition.
+	if len(deploy.Status.Conditions) > 0 {
+		filteredConditions := make([]appsv1.DeploymentCondition, 0)
+		for _, cond := range deploy.Status.Conditions {
+			if cond.Type == dvcr_deployment_condition.MaintenanceType {
+				if cond.Reason != condition.Reason || cond.Message != condition.Message {
+					condition.LastTransitionTime = metav1.Now()
+				}
+				cond = condition
+			}
+			// Copy non-maintenance conditions.
+			filteredConditions = append(filteredConditions, cond)
+		}
+		deploy.Status.Conditions = filteredConditions
+	}
+
+	// Deploy has no conditions, create new slice.
+	deploy.Status.Conditions = []appsv1.DeploymentCondition{condition}
```

Comment:
It probably should be in the `else` statement, otherwise it will always overwrite conditions which are handled previously